src/modules/featurepopup/featurepopup-directive.js

```diff
@@ -214,8 +216,14 @@ angular.module('anol.featurepopup')
                                     return;
                                 }
 
-                                var _features = layer.olLayer.getSource()
-                                    .getFeaturesInExtent(extent)
+                                let _featuresInExtent;
+                                if (layer.olLayer.getSource() instanceof VectorTile) {
+                                  _featuresInExtent = layer.olLayer.getFeaturesInExtent(extent);
+                                } else {
+                                  _featuresInExtent = layer.olLayer.getSource()
+                                    .getFeaturesInExtent(extent);
+                                }
```

Comment:
```suggestion
                                if (layer.olLayer instanceof VectorTileLayer) {
                                  _featuresInExtent = layer.olLayer.getFeaturesInExtent(extent);
                                } else if (layer.olLayer.getSource() instanceof VectorSource) {
                                  _featuresInExtent = layer.olLayer.getSource()
                                    .getFeaturesInExtent(extent);
                                }
```
I would prefer if the if would check for the class that actually contains `getFeaturesInExtent`, if that works.

The `else if` is probably not necessary.

Reply:
done